pkg/cmd/codespace/ssh.go

```diff
@@ -449,27 +412,39 @@ func getConfiguredPrivateKeys(
 		return nil, fmt.Errorf("could not load ssh configuration: %w", err)
 	}
 
-	var privateKeyPaths []string
-
 	userHomeDir, _ := os.UserHomeDir()
 
 	configLines := strings.Split(string(configBytes), "\n")
 	for _, line := range configLines {
 		line = strings.TrimSpace(line)
 
 		if strings.HasPrefix(line, "identityfile ") {
-			path := strings.SplitN(line, " ", 2)[1]
+			privateKeyPath := strings.SplitN(line, " ", 2)[1]
 
-			if strings.HasPrefix(path, "~") {
+			if strings.HasPrefix(privateKeyPath, "~") {
 				// os.Stat can't handle ~, so convert it to the real path
-				path = strings.Replace(path, "~", userHomeDir, 1)
+				privateKeyPath = strings.Replace(privateKeyPath, "~", userHomeDir, 1)
 			}
 
-			privateKeyPaths = append(privateKeyPaths, path)
+			// The default configuration includes standard keys like id_rsa or id_ed25519,
+			// but these may not actually exist
+			if _, err := os.Stat(privateKeyPath); err != nil {
+				continue
+			}
+
+			publicKeyPath := privateKeyPath + ".pub"
+			if _, err := os.Stat(publicKeyPath); err != nil {
+				continue
+			}
+
+			return &ssh.KeyPair{
+				PrivateKeyPath: privateKeyPath,
+				PublicKeyPath:  publicKeyPath,
+			}, nil
 		}
 	}
 
-	return privateKeyPaths, nil
+	return nil, nil
```

Comment:
This is a bit of a headscratch. More idiomatic in Go would be to return some sort of sentinel error indicating none was found to make the behavior explicit. Callers of this will need to understand that double nil is possible and that it needs to be checked.

Reply:
I'm ok to add a sentinel error. FWIW, if we add a sentinel error doesn't that just mean that callers need to be aware of the sentinel error instead of the double `nil`? 😉

Reply:
Yep exactly, but everyone in Go is used to checking:
```
if err != nil
```

And after that expected the valid case to not give you a nil value. Yes, they will need to do handle the not found error, but that's in line with out `os.Stat`, `db.QueryRow` and other Go apis work.

Reply:
Updated